Rename AbstractFrame to Frame

cdumez opened the pull request:
#### 7651b3eb50d3e41d859357345a2c1759b6a4659b
<pre>
Rename AbstractFrame to Frame
<a href="https://bugs.webkit.org/show_bug.cgi?id=253853">https://bugs.webkit.org/show_bug.cgi?id=253853</a>

Reviewed by Darin Adler.

Rename AbstractFrame to Frame, now that the subclasses are LocalFrame and
RemoteFrame.

* Source/WebCore/*:
* Source/WebKit/*:
* Source/WebKitLegacy/*:

Canonical link: <a href="https://commits.webkit.org/261612@main">https://commits.webkit.org/261612@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9a66f0ce5cfddd823e0744690cf0c4641e159cc3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/112315 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/21454 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/950 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/4092 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/120928 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/116379 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/22792 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/12507 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/86/builds/4866 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/118080 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/16935 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/100117 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/105361 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/98892 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/88/builds/655 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/45942 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/13827 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/89/builds/703 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/94334 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/14519 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/9989 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/19870 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/52708 "Passed tests") | | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/74/builds/8094 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/16323 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/4409 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/dom/DeviceOrientationAndMotionAccessController.cpp:
```diff
@@ -82,7 +82,7 @@ void DeviceOrientationAndMotionAccessController::shouldAllowAccess(const Documen
             return;
 
-        for (AbstractFrame* frame = m_topDocument.frame(); frame && frame->window(); frame = frame->tree().traverseNext()) {
+        for (Frame* frame = m_topDocument.frame(); frame && frame->window(); frame = frame->tree().traverseNext()) {
```
Can this use `auto*`?

achristensen07 replied:
This one can't.  Document::frame returns the class that used to be called Frame and is now LocalFrame, but in this loop we need it to be the class that used to be called AbstractFrame and is now Frame.

darinadler commented on Source/WebCore/dom/Document.cpp:
```diff
@@ -466,7 +466,7 @@ static bool canAccessAncestor(const SecurityOrigin& activeSecurityOrigin, LocalF
 
     const bool isLocalActiveOrigin = activeSecurityOrigin.isLocal();
-    for (RefPtr<AbstractFrame> ancestorFrame = targetFrame; ancestorFrame; ancestorFrame = ancestorFrame->tree().parent()) {
+    for (RefPtr<Frame> ancestorFrame = targetFrame; ancestorFrame; ancestorFrame = ancestorFrame->tree().parent()) {
```
Can this use just `RefPtr`?

darinadler commented on Source/WebCore/dom/Document.cpp:
```diff
@@ -6220,7 +6220,7 @@ void Document::setTransformSource(std::unique_ptr<TransformSource> source)
 {
     m_designMode = value;
-    for (RefPtr<AbstractFrame> frame = m_frame.get(); frame; frame = frame->tree().traverseNext(m_frame.get())) {
+    for (RefPtr<Frame> frame = m_frame.get(); frame; frame = frame->tree().traverseNext(m_frame.get())) {
```
Can this use just `RefPtr`?

darinadler commented on Source/WebCore/dom/UserGestureIndicator.cpp:
```diff
@@ -69,7 +69,7 @@ UserGestureToken::UserGestureToken(ProcessingUserGestureState state, UserGesture
 
     auto& documentOrigin = document->securityOrigin();
-    for (AbstractFrame* frame = &documentFrame->tree().top(); frame; frame = frame->tree().traverseNext()) {
+    for (Frame* frame = &documentFrame->tree().top(); frame; frame = frame->tree().traverseNext()) {
```
Can this use `auto*`?

darinadler commented on Source/WebCore/dom/UserGestureIndicator.cpp:
```diff
@@ -116,7 +116,7 @@ UserGestureIndicator::UserGestureIndicator(std::optional<ProcessingUserGestureSt
         if (auto* frame = document->frame()) {
             if (!frame->hasHadUserInteraction()) {
-                for (AbstractFrame *ancestor = frame; ancestor; ancestor = ancestor->tree().parent()) {
+                for (Frame *ancestor = frame; ancestor; ancestor = ancestor->tree().parent()) {
```
Can this use `auto*`?

darinadler commented on Source/WebCore/html/HTMLFrameOwnerElement.cpp:
```diff
@@ -136,7 +136,7 @@ bool HTMLFrameOwnerElement::isProhibitedSelfReference(const URL& completeURL) co
     // We allow one level of self-reference because some websites depend on that, but we don't allow more than one.
     bool foundOneSelfReference = false;
-    for (AbstractFrame* frame = document().frame(); frame; frame = frame->tree().parent()) {
+    for (Frame* frame = document().frame(); frame; frame = frame->tree().parent()) {
```
Can this use `auto*`?

darinadler commented on Source/WebCore/inspector/PageDebugger.cpp:
```diff
@@ -148,7 +148,7 @@ void PageDebugger::reportException(JSGlobalObject* state, JSC::Exception* except
 {
     for (auto& page : pageGroup.pages()) {
-        for (AbstractFrame* frame = &page.mainFrame(); frame; frame = frame->tree().traverseNext()) {
+        for (Frame* frame = &page.mainFrame(); frame; frame = frame->tree().traverseNext()) {
```
Can this use `auto*`?

darinadler commented on Source/WebCore/inspector/agents/InspectorPageAgent.cpp:
```diff
@@ -307,7 +307,7 @@ Protocol::Page::ResourceType InspectorPageAgent::cachedResourceTypeJSON(const Ca
 LocalFrame* InspectorPageAgent::findFrameWithSecurityOrigin(Page& page, const String& originRawString)
 {
-    for (AbstractFrame* frame = &page.mainFrame(); frame; frame = frame->tree().traverseNext()) {
+    for (Frame* frame = &page.mainFrame(); frame; frame = frame->tree().traverseNext()) {
```
Can this use `auto*`?

darinadler commented on Source/WebCore/inspector/agents/InspectorPageAgent.cpp:
```diff
@@ -619,7 +619,7 @@ Protocol::ErrorStringOr<Ref<JSON::ArrayOf<Protocol::Page::Cookie>>> InspectorPag
     ListHashSet<Cookie> allRawCookies;
 
-    for (AbstractFrame* frame = &m_inspectedPage.mainFrame(); frame; frame = frame->tree().traverseNext()) {
+    for (Frame* frame = &m_inspectedPage.mainFrame(); frame; frame = frame->tree().traverseNext()) {
```
Can this use `auto*`?

darinadler commented on Source/WebCore/inspector/agents/InspectorPageAgent.cpp:
```diff
@@ -730,7 +730,7 @@ Protocol::ErrorStringOr<void> InspectorPageAgent::setCookie(Ref<JSON::Object>&&
         return makeUnexpected(errorString);
 
-    for (AbstractFrame* frame = &m_inspectedPage.mainFrame(); frame; frame = frame->tree().traverseNext()) {
+    for (Frame* frame = &m_inspectedPage.mainFrame(); frame; frame = frame->tree().traverseNext()) {
```
Can this use `auto*`?

darinadler reviewed (approved):
I stopped commenting about all the `auto*` places.

cdumez committed:
Rename AbstractFrame to Frame
https://bugs.webkit.org/show_bug.cgi?id=253853

Reviewed by Darin Adler.

Rename AbstractFrame to Frame, now that the subclasses are LocalFrame and
RemoteFrame.

* Source/WebCore/*:
* Source/WebKit/*:
* Source/WebKitLegacy/*:

Canonical link: https://commits.webkit.org/261612@main
